### PR TITLE
feat(pipeline): Change the compensation policy not executed by cron

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
Change the compensation policy not executed by cron, and change the policy of the latest one after successful execution to the latest one after execution termination

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTY0XSwiYXNzaWduZWUiOlsiMTAwMDU2MCJdfQ%3D%3D&id=230962&iterationID=1164&pId=0&type=REQUIREMENT)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       Change the compensation policy not executed by cron, and change the policy of the latest one after successful execution to the latest one after execution termination       |
| 🇨🇳 中文    |        更改 cron 未执行补偿策略，将执行成功往后最新一条的策略改成执行终止状态往后最新一条      |

### CODE REVIEW

File: modules/pipeline/providers/cron/compensator/provider.go [hunk 1]
```diff
@@ -357,10 +357,16 @@ func (p *provider) doCronCompensate(compensator apistructs.CronCompensator, notR
 	// it should be compared with the pipeline in the latest success status. Only the ID greater than the successful ID can be executed
 	if compensator.LatestFirst && compensator.StopIfLatterExecuted {
 		// Get the pipeline successfully executed
-		runSuccessPipeline, _, _, _, err := p.dbClient.PageListPipelines(apistructs.PipelinePageListRequest{
+		endStatusPipelines, _, _, _, err := p.dbClient.PageListPipelines(apistructs.PipelinePageListRequest{
```
Comment (Effet) [inline modules/pipeline/providers/cron/compensator/provider.go:360]: too many returns, can wrap a struct

File: modules/pipeline/dbclient/op_pipeline.go [hunk 2]
```diff
@@ -473,10 +482,19 @@ func (client *Client) PageListPipelines(req apistructs.PipelinePageListRequest,
...
 	pipelines, err := client.ListPipelinesByIDs(pagingPipelineIDs, needQueryDefinition, ops...)
 	if err != nil {
-		return nil, pagingPipelineIDs, -1, -1, err
-	}
-
-	return pipelines, pagingPipelineIDs, total, currentPageSize, nil
+		return &PageListPipelinesResult{
+			PagingPipelineIDs: pagingPipelineIDs,
+			Total:             -1,
+			CurrentPageSize:   -1,
```
Comment (Effet) [inline modules/pipeline/dbclient/op_pipeline.go:488]: what means -1? why not use default value 0?

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/pipeline/dbclient/op_pipeline.go (hunk 2 above)
Comment (kakj-go) [reply modules/pipeline/dbclient/op_pipeline.go:488]: I don't know. The old logic. I just changed the lower structure

Comment (Effet) [reply modules/pipeline/dbclient/op_pipeline.go:488]: then, you should to know it. you are team member responsible for the pipeline.

**Review by sfwn — APPROVED** [review]

File: modules/pipeline/dbclient/op_pipeline.go (hunk 2 above)
Comment (kakj-go) [reply modules/pipeline/dbclient/op_pipeline.go:488]: ok

**Review by sfwn — APPROVED** [review]

**Pull request merged**